generator/plugins/proto2gql/dataloaders.go

```diff
@@ -0,0 +1,293 @@
+package proto2gql
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/EGT-Ukraine/go2gql/generator/plugins/dataloader"
+	"github.com/EGT-Ukraine/go2gql/generator/plugins/graphql"
+	"github.com/EGT-Ukraine/go2gql/generator/plugins/graphql/lib/importer"
+	"github.com/EGT-Ukraine/go2gql/generator/plugins/proto2gql/parser"
+)
+
+func (g Proto2GraphQL) registerMethodDataLoaders(sc ServiceConfig, cfg MethodConfig, file *parsedFile, method *parser.Method) error {
+	for name, cfg := range cfg.DataLoaderProvider {
+		if err := g.registerMethodDataLoader(name, cfg, sc, file, method); err != nil {
+			return errors.Wrapf(err, "failed to register %s data loader", name)
+		}
+	}
+
+	return nil
+}
+
+func (g Proto2GraphQL) registerMethodDataLoader(name string, cfg DataLoaderConfig, serviceConfig ServiceConfig, file *parsedFile, method *parser.Method) error {
+	err := g.validateDataLoader(cfg, method)
+	if err != nil {
+		return errors.Wrap(err, "validation error")
+	}
+
+	resultField, err := messageFieldByPath(method.OutputMessage, cfg.ResultField)
+	if err != nil {
+		return errors.Wrap(err, "failed to get result field")
+	}
+	normalResultField := resultField.(*parser.NormalField)
+	resultMessage := normalResultField.Type.(*parser.Message)
+
+	matchField, err := messageFieldByPath(resultField.GetType().(*parser.Message), cfg.MatchField)
+	if err != nil {
+		return errors.Wrap(err, "failed to get match field")
+	}
+
+	responseGoType, err := g.goTypeByParserType(normalResultField.Type)
+	if err != nil {
+		return errors.Wrap(err, "failed to get result field go type")
+	}
+	outputMsgTypeFile, err := g.parsedFile(method.OutputMessage.File())
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve file type file")
+	}
+
+	dataLoaderOutType, err := g.TypeOutputTypeResolver(outputMsgTypeFile, resultMessage)
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve output type")
+	}
+
+	var fetchCode func(importer *importer.Importer) string
+	if cfg.Type == DataLoaderType1ToN {
```

Comment:
Maybe we should add cfg.Type validation?

Reply:
ready

Reply:
I validate it earlier